### PR TITLE
Credential Scope Configuration Test

### PR DESCRIPTION
**Pull request by davidh44**
## Modifications

Test that `credentialScope` configuration precedence is honored in the following order: request -> client -> env var -> profile

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/aws/aws-sdk-java-v2/blob/master/CONTRIBUTING.md) document
- [ ] Local run of `mvn install` succeeds
- [ ] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] I have added a changelog entry. Adding a new entry must be accomplished by running the `scripts/new-change` script and following the instructions. Commit the new file created by the script in `.changes/next-release` with your changes.
- [ ] My change is to implement 1.11 parity feature and I have updated [LaunchChangelog](https://github.com/aws/aws-sdk-java-v2/blob/master/docs/LaunchChangelog.md)

## License

- [ ] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

File: test/codegen-generated-classes-test/src/test/java/software/amazon/awssdk/services/CredentialScopeTest.java [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+        capturingInterceptor = new CapturingInterceptor();
+        clientBuilder = ProtocolQueryClient.builder()
+                                    .overrideConfiguration(o -> o.addExecutionInterceptor(capturingInterceptor))
+                                    .endpointOverride(URI.create("http://localhost:" + wiremock.getHttpPort()));
+    }
+
+    @AfterEach
+    public void reset() {
+        clearSystemProperties();
+    }
```
Comment (millems) [inline test/codegen-generated-classes-test/src/test/java/software/amazon/awssdk/services/CredentialScopeTest.java:69]: Instead of clearing after, it's better to save before and restore after, in case someone is using these system properties locally for integration tests in the same run. Otherwise the integration tests may pass when run alone, but fail after this test is run.

File: test/codegen-generated-classes-test/src/test/java/software/amazon/awssdk/services/CredentialScopeTest.java [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+
+    /**
+     * Returns all possible Boolean combinations
+     */
+    private static Stream<Arguments> testParams() {
+        return IntStream.range(0, 1 << 4)
+                        .mapToObj(i -> Arguments.of((i & 1) == 1, (i & 2) == 2, (i & 4) == 4, (i & 8) == 8));
```
Comment (millems) [inline test/codegen-generated-classes-test/src/test/java/software/amazon/awssdk/services/CredentialScopeTest.java:76]: Cute.

File: test/codegen-generated-classes-test/src/test/java/software/amazon/awssdk/services/CredentialScopeTest.java [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+    private void clearSystemProperties() {
+        System.clearProperty(SdkSystemSetting.AWS_ACCESS_KEY_ID.property());
+        System.clearProperty(SdkSystemSetting.AWS_SECRET_ACCESS_KEY.property());
+        System.clearProperty(SdkSystemSetting.AWS_CREDENTIAL_SCOPE.property());
+    }
+
+    private ProfileFile credentialFile(String name, String accessKeyId, String secretAccessKey, String credentialScope) {
+        String contents = String.format("[%s]\naws_access_key_id = %s\naws_secret_access_key = %s\naws_credential_scope = %s",
+                                        name, accessKeyId, secretAccessKey, credentialScope);
```
Comment (millems) [inline test/codegen-generated-classes-test/src/test/java/software/amazon/awssdk/services/CredentialScopeTest.java:139]: Is a blank `aws_credential_scope` a case we need to test? or is that treated the same as null for priority purposes?

**Review by millems — APPROVED** [review]
